lib/mongo/client.rb

```diff
@@ -443,7 +443,12 @@ def initialize(addresses_or_uri, options = nil)
         @srv_records = uri.srv_records
       else
         addresses = addresses_or_uri
-        @srv_records = nil
+        if addresses.any? { |addr| addr =~ /mongodb(\+srv)?:\/\// }
+          raise "Protocol should not be included in host list. Did you mean to not use an array?"
```

Comment:
I think this line can raise `ArgumentError` going by the validation  methods toward the bottom of the file.

The offending address should be included in the exception message, which means probably the raising should be done inside the loop that iterates the addresses.